cedar-policy-formatter/src/pprint/fmt.rs

```diff
@@ -135,6 +143,49 @@ mod tests {
 
     use super::*;
 
+    #[test]
+    fn test_soundness_check() {
+        let p1 = r#"permit (principal, action, resource)
+        when { "
+        
+        a
+        " };"#;
+        let p2 = r#"permit (principal, action, resource)
+        when { "
+        a
+        " };"#;
+        assert!(soundness_check(p2, &parse_policyset(p1).unwrap()).is_err());
```

Comment:
Probably a dumb question, but why does the change in this PR help identify the issue in #862? Looking at the diff, it seems like the only change is to add a comparison of policy ids, which doesn't seem related.

Or is this test case unrelated to the changed code?

Reply:
`templates()` to `policies()` I think

Reply:
Aha, thanks. @shaobo-he-aws can you add a description to the PR so it's easy not to miss this.

Reply:
Done.